### ISSUE TITLE
Docker won't start when switching to windows containers

### ISSUE THREAD

**Issue by AbsurdusAdeptus** [docker/for-win#711]
### Expected behavior
Docker starts...

### Actual behavior
Docker fails to start for Windows containers. Showing dialog saying:
Docker daemon failed with message:
time="2017-05-07T19:04:36-07:00" level=warning msg="Running experimental build" 
time="2017-05-07T19:04:36.360059800-07:00" level=debug msg="Listener created for HTTP on npipe (//./pipe/docker_engine_windows)" 
time="2017-05-07T19:04:36.365056800-07:00" level=info msg="Windows default isolation mode: hyperv" 
time="2017-05-07T19:04:36.365056800-07:00" level=debug msg="Using default logging driver json-file" 
time="2017-05-07T19:04:36.367058400-07:00" level=debug msg="WindowsGraphDriver InitFilter at C:\\ProgramData\\Docker\\windowsfilter" 
time="2017-05-07T19:04:36.367058400-07:00" level=debug msg="Using graph driver windowsfilter" 
time="2017-05-07T19:04:36.367058400-07:00" level=debug msg="Max Concurrent Downloads: 3" 
time="2017-05-07T19:04:36.367058400-07:00" level=debug msg="Max Concurrent Uploads: 5" 
time="2017-05-07T19:04:36.401752600-07:00" level=info msg="Graph migration to content-addressability took 0.00 seconds" 
time="2017-05-07T19:04:36.402753600-07:00" level=info msg="Loading containers: start." 
time="2017-05-07T19:04:36.402753600-07:00" level=debug msg="Option Experimental: true" 
time="2017-05-07T19:04:36.402753600-07:00" level=debug msg="Option DefaultDriver: nat" 
time="2017-05-07T19:04:36.402753600-07:00" level=debug msg="Option DefaultNetwork: nat" 
time="2017-05-07T19:04:36.545747700-07:00" level=debug msg="Network Response : [ { \"ComputedPolicies\" : [ { } ], \"CurrentEndpointCount\" : 0, \"ID\" : \"eeb47cdc-ff82-4447-8276-a6a5b5a1427d\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-1E-6F-FF\", \"StartMacAddress\" : \"00-15-5D-1E-60-00\" } ], \"MaxConcurrentEndpoints\" : 0, \"Name\" : \"DockerNAT (5)\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.30.0.0/20\" } ], \"TotalEndpoints\" : 0, \"Type\" : \"Private\", \"Version\" : 4294967297 }, { \"ComputedPolicies\" : [ { } ], \"CurrentEndpointCount\" : 0, \"ID\" : \"a755468f-03f5-414b-9374-5f17ae7b4178\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-34-0F-FF\", \"StartMacAddress\" : \"00-15-5D-34-00-00\" } ], \"MaxConcurrentEndpoints\" : 0, \"Name\" : \"DockerNAT (3)\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.28.112.0/20\" } ], \"TotalEndpoints\" : 0, \"Type\" : \"Private\", \"Version\" : 4294967297 }, { \"CurrentEndpointCount\" : 0, \"ID\" : \"e6839ed8-03f6-44d6-8d73-93708629aa1b\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-99-DF-FF\", \"StartMacAddress\" : \"00-15-5D-99-D0-00\" } ], \"MaxConcurrentEndpoints\" : 0, \"Name\" : \"DockerNAT (1)\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.17.144.0/20\" } ], \"TotalEndpoints\" : 0, \"Type\" : \"Private\", \"Version\" : 4294967297 }, { \"ComputedPolicies\" : [ { } ], \"CurrentEndpointCount\" : 0, \"ID\" : \"80f69614-4d1d-45b3-938d-431929d708bc\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-73-DF-FF\", \"StartMacAddress\" : \"00-15-5D-73-D0-00\" } ], \"MaxConcurrentEndpoints\" : 0, \"Name\" : \"DockerNAT\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.27.128.0/20\" } ], \"TotalEndpoints\" : 0, \"Type\" : \"Private\", \"Version\" : 4294967297 }, { \"CurrentEndpointCount\" : 0, \"Degraded\" : true, \"ExternalInterfaceAddress\" : \"00-15-5D-F7-69-81\", \"ExternalInterfaceIndex\" : 22, \"ID\" : \"b4a7fdc3-0ba5-4830-8a1c-3cec36608b6b\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-F7-6F-FF\", \"StartMacAddress\" : \"00-15-5D-F7-60-00\" } ], \"MaxConcurrentEndpoints\" : 1, \"NATName\" : \"Hb4a7fdc3-0ba5-4830-8a1c-3cec36608b6b\", \"Name\" : \"nat\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.30.176.0/20\", \"GatewayAddress\" : \"172.30.176.1\" } ], \"SwitchGuid\" : \"853e1840-84e6-4cb1-8eb4-baf8d8e60771\", \"TotalEndpoints\" : 3, \"Type\" : \"nat\", \"Version\" : 4294967297 }, { \"ComputedPolicies\" : [ { } ], \"CurrentEndpointCount\" : 0, \"ID\" : \"ca9ef76b-ac6e-45c7-8981-e502cd69068c\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-AB-EF-FF\", \"StartMacAddress\" : \"00-15-5D-AB-E0-00\" } ], \"MaxConcurrentEndpoints\" : 0, \"Name\" : \"DockerNAT (2)\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.31.128.0/20\" } ], \"TotalEndpoints\" : 0, \"Type\" : \"Private\", \"Version\" : 4294967297 }, { \"ComputedPolicies\" : [ { } ], \"CurrentEndpointCount\" : 0, \"ID\" : \"9aa5aaeb-b1e6-494c-a5f5-c4485521d980\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-76-9F-FF\", \"StartMacAddress\" : \"00-15-5D-76-90-00\" } ], \"MaxConcurrentEndpoints\" : 0, \"Name\" : \"DockerNAT (4)\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.29.176.0/20\" } ], \"TotalEndpoints\" : 0, \"Type\" : \"Private\", \"Version\" : 4294967297 } ]" 
time="2017-05-07T19:04:36.557748900-07:00" level=debug msg="Launching DNS server for network%!(EXTRA string=none)" 
time="2017-05-07T19:04:36.585749000-07:00" level=warning msg="Unable to locate plugin: private, retrying in 1s" 
time="2017-05-07T19:04:37.585777300-07:00" level=warning msg="Unable to locate plugin: private, retrying in 2s" 
time="2017-05-07T19:04:39.585820700-07:00" level=warning msg="Unable to locate plugin: private, retrying in 4s" 
time="2017-05-07T19:04:43.586466100-07:00" level=warning msg="Unable to locate plugin: private, retrying in 8s" 
time="2017-05-07T19:04:51.587151200-07:00" level=error msg="Error occurred when creating network plugin not found" 
time="2017-05-07T19:04:51.598052500-07:00" level=warning msg="Unable to locate plugin: private, retrying in 1s" 
time="2017-05-07T19:04:52.598598200-07:00" level=warning msg="Unable to locate plugin: private, retrying in 2s" 
time="2017-05-07T19:04:54.599244300-07:00" level=warning msg="Unable to locate plugin: private, retrying in 4s" 
time="2017-05-07T19:04:58.601015000-07:00" level=warning msg="Unable to locate plugin: private, retrying in 8s" 
time="2017-05-07T19:05:06.601947100-07:00" level=error msg="Error occurred when creating network plugin not found" 
time="2017-05-07T19:05:06.620792200-07:00" level=warning msg="Unable to locate plugin: private, retrying in 1s" 
time="2017-05-07T19:05:07.621898700-07:00" level=warning msg="Unable to locate plugin: private, retrying in 2s" 
time="2017-05-07T19:05:09.624899700-07:00" level=warning msg="Unable to locate plugin: private, retrying in 4s" 
time="2017-05-07T19:05:13.625084400-07:00" level=warning msg="Unable to locate plugin: private, retrying in 8s" 
time="2017-05-07T19:05:21.625486000-07:00" level=error msg="Error occurred when creating network plugin not found" 
time="2017-05-07T19:05:21.664279400-07:00" level=warning msg="Unable to locate plugin: private, retrying in 1s" 
time="2017-05-07T19:05:22.664642900-07:00" level=warning msg="Unable to locate plugin: private, retrying in 2s" 
time="2017-05-07T19:05:24.666379100-07:00" level=warning msg="Unable to locate plugin: private, retrying in 4s" 
time="2017-05-07T19:05:28.667852400-07:00" level=warning msg="Unable to locate plugin: private, retrying in 8s" 
time="2017-05-07T19:05:36.669880300-07:00" level=error msg="Error occurred when creating network plugin not found" 
time="2017-05-07T19:05:36.748867400-07:00" level=debug msg="releasing IPv4 pools from network nat (09553edeab3ec686164c3be0e55fc758cd2fc04a485e552b2b12b73abfc85db0)" 
time="2017-05-07T19:05:36.748867400-07:00" level=debug msg="ReleaseAddress(172.30.176.0/20, 172.30.176.1)" 
time="2017-05-07T19:05:36.748867400-07:00" level=debug msg="ReleasePool(172.30.176.0/20)" 
time="2017-05-07T19:05:36.805569800-07:00" level=debug msg="Allocating IPv4 pools for network nat (1c43272d96b977863a2063dd67835f43b9ad93b9a65bdfe9fce773ba6b8eaffb)" 
time="2017-05-07T19:05:36.805569800-07:00" level=debug msg="RequestPool(LocalDefault, 172.30.176.0/20, , map[], false)" 
time="2017-05-07T19:05:36.805569800-07:00" level=debug msg="RequestAddress(172.30.176.0/20, 172.30.176.1, map[RequestAddressType:com.docker.network.gateway])" 
time="2017-05-07T19:05:36.805569800-07:00" level=debug msg="Launching DNS server for network%!(EXTRA string=nat)" 
time="2017-05-07T19:05:36.813540200-07:00" level=debug msg="Network Response : { \"CurrentEndpointCount\" : 0, \"Degraded\" : true, \"ExternalInterfaceAddress\" : \"00-15-5D-F7-69-81\", \"ExternalInterfaceIndex\" : 22, \"ID\" : \"b4a7fdc3-0ba5-4830-8a1c-3cec36608b6b\", \"MacPools\" : [ { \"EndMacAddress\" : \"00-15-5D-F7-6F-FF\", \"StartMacAddress\" : \"00-15-5D-F7-60-00\" } ], \"MaxConcurrentEndpoints\" : 1, \"NATName\" : \"Hb4a7fdc3-0ba5-4830-8a1c-3cec36608b6b\", \"Name\" : \"nat\", \"Policies\" : [  ], \"Subnets\" : [ { \"AddressPrefix\" : \"172.30.176.0/20\", \"GatewayAddress\" : \"172.30.176.1\" } ], \"SwitchGuid\" : \"853e1840-84e6-4cb1-8eb4-baf8d8e60771\", \"TotalEndpoints\" : 3, \"Type\" : \"nat\", \"Version\" : 4294967297 }" 
time="2017-05-07T19:05:36.813540200-07:00" level=debug msg="Binding a resolver on network nat gateway 172.30.176.1" 
time="2017-05-07T19:05:36.814543300-07:00" level=error msg="Resolver Setup/Start failed for container nat, \"error in opening name server socket listen udp 172.30.176.1:53: bind: The requested address is not valid in its context.\"" 

   at Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings)
   at Docker.Backend.ContainerEngine.Windows.Start(Settings settings)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)

### Information
30EEA893-1F17-4DFF-A760-FD396C780869/2017-05-07_18-56-50

Windows 10 pro fully updated.

Docker Version 17.03.1-ce-win5 (10743)
Channel: stable
b18e2a5

### Steps to reproduce the behavior

  1. Installed docker.
  2. Waited for docker to startup and run.
  3. Right clicked on the whale icon and selected "Switch to Windows containers..."
  4. Waited for whale icon stop moving.

**Comment by AbsurdusAdeptus** [comment]
Does anyone have any ideas? I'm at a complete loss here. I've tried everything I can think of short of refreshing my windows and I really wouldn't want to have to do that.

**Comment by AbsurdusAdeptus** [comment]
Ok... I got it working. I'm not quite sure what really got it working, but I tried a few more things. I again dissabled hyper-v and containers and let docker re-enable them. It still wasn't working after that. So I found this issue: https://github.com/moby/moby/issues/31469 and tried a recomendation by @msabansal to:

net stop hns;
del c:\programdata\Microsoft\windows\HNS.data
Remove-Netnat;
net start hns;

The delete command failed with a file doesn't exist, but when I navigated there in explorer the file was there, so I deleted it through that and ran the other two commands. I tried to shutdown the docker that was in a failed startup state, but it just started saying it would shutdown after it was done (with what I don't know, it was red and had failed to start)... I manually killed the docker process, as well as the docker for windows service process. When I restarted the docker process it told me the service was off and it needed permissions to re-enable it, when I hit yest on the ensuing UAC docker started up just fine... I assume it was the deleting of the hns data file that solved my issue, because I had dont almost all of the other things before.
